From 94a9870932e1a145275ebf1ea3aa2c515c91aad0 Mon Sep 17 00:00:00 2001
From: Snehit Sah <snehitsah@protonmail.com>
Date: Mon, 24 Jan 2022 12:21:16 +0530
Subject: [PATCH] Update settings panel for Flatpak

Cannot set custom file manager when running KDF as Flatpak
so disable file manager selection widgets and display a
message below them.

Signed-off-by: Snehit Sah <snehitsah@protonmail.com>
---
 src/kdfconfig.cpp |  7 +++++++
 src/kdfconfig.ui  | 15 +++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/src/kdfconfig.cpp b/src/kdfconfig.cpp
index 9ce0a31..3ce6e1d 100644
--- a/src/kdfconfig.cpp
+++ b/src/kdfconfig.cpp
@@ -79,6 +79,13 @@ KDFConfigWidget::KDFConfigWidget(QWidget *parent, bool init)
         connect(mOpenMountCheck,&QAbstractButton::toggled,this,&KDFConfigWidget::slotChanged);
 
         connect(mPopupFullCheck,&QAbstractButton::toggled,this,&KDFConfigWidget::slotChanged);
+
+        if(!QStandardPaths::locate(QStandardPaths::RuntimeLocation, QLatin1String("flatpak-info")).isEmpty())
+        {
+            mFlatpakLabel->setVisible( true );
+            mSystemFileManagerCheck->setEnabled( false );
+            mFileManagerEdit->setEnabled( false );
+        }
     }
 
     loadSettings();
diff --git a/src/kdfconfig.ui b/src/kdfconfig.ui
index d68d16c..810128f 100644
--- a/src/kdfconfig.ui
+++ b/src/kdfconfig.ui
@@ -80,6 +80,21 @@
    <item row="3" column="1">
     <widget class="KLineEdit" name="mFileManagerEdit"/>
    </item>
+   <item row="4" column="0" colspan="2">
+    <widget class="QLabel" name="mFlatpakLabel">
+     <property name="visible">
+      <bool>false</bool>
+     </property>
+     <property name="font">
+      <font>
+       <italic>true</italic>
+      </font>
+     </property>
+     <property name="text">
+      <string>Custom file manager not supported when running as Flatpak.</string>
+     </property>
+    </widget>
+   </item>
   </layout>
  </widget>
  <customwidgets>
-- 
2.34.1

